python/cudf/cudf/core/indexing.py

```diff
@@ -266,11 +266,22 @@ def _getitem_tuple_arg(self, arg):
             else:
                 return columns_df.index._get_row_major(columns_df, arg[0])
         else:
-            df = DataFrame()
-            for col in columns_df.columns:
-                # need Series() in case a scalar is returned
-                df[col] = Series(columns_df[col].loc[arg[0]])
-            df.columns = columns_df.columns
+            if hasattr(arg[0], "__len__"):
+                arg = (column.as_column(arg[0]), arg[1])
+
+            if pd.api.types.is_bool_dtype(arg[0]):
+                df = columns_df.take(arg[0])
+            elif isinstance(arg[0], column.ColumnBase):
+                indices = (
+                    columns_df[columns_df.columns[0]].loc._loc_to_iloc(arg[0])
+                )
```

Comment:
Isn't this just a join against the index column(s)? We shouldn't need to convert to a gather map and then run the gather, we should be able to just do the join. Apologies if I mislead you with a previous review comment.

Reply:
Connected offline, we can't just use a join here as things like slices should still be done zero copy if possible.